actor/builtin/paymentbroker/paymentbroker.go

```diff
@@ -334,6 +341,49 @@ func (pb *Actor) Extend(vmctx exec.VMContext, chid *types.ChannelID, eol *types.
 	return 0, nil
 }
 
+// Cancel can be used to end a storage deal early. It lowers the EOL of the
+// payment channel to 1 blocktime from now and allows a client to reclaim their
+// payments. In the time before the channel is closed, a miner can potentially
+// dispute a closer.
+func (pb *Actor) Cancel(vmctx exec.VMContext, chid *types.ChannelID) (uint8, error) {
+	if err := vmctx.Charge(actor.DefaultGasCost); err != nil {
+		return exec.ErrInsufficientGas, errors.RevertErrorWrap(err, "Insufficient gas")
+	}
+
+	ctx := context.Background()
+	storage := vmctx.Storage()
+	payerAddress := vmctx.Message().From
+
+	err := withPayerChannels(ctx, storage, payerAddress, func(byChannelID exec.Lookup) error {
+		chInt, err := byChannelID.Find(ctx, chid.KeyString())
+		if err != nil {
+			if err == hamt.ErrNotFound {
+				return Errors[ErrUnknownChannel]
+			}
+			return errors.FaultErrorWrapf(err, "Could not retrieve payment channel with ID: %s", chid)
+		}
+
+		channel, ok := chInt.(*PaymentChannel)
+		if !ok {
+			return errors.NewFaultError("Expected PaymentChannel from channels lookup")
+		}
+
+		channel.Eol = vmctx.BlockHeight().Add(types.NewBlockHeight(1))
```

Comment:
BLOCKING: We're not exactly sure how much to add here, but it isn't 1. 10000 is around 3 days and seems reasonable for now. This needs to be a constant and it needs an issue to determine the correct value. #1887 is a good template.